src/renderers/dom/shared/__tests__/DOMPropertyOperations-test.js

```diff
@@ -261,4 +261,20 @@ describe('DOMPropertyOperations', () => {
       );
     });
   });
+
+  describe('getValueForProperty', function() {
+    it('uses getValueForAttribute when given a custom attribute', function() {
+      var node = document.createElement('div');
+
+      node.setAttribute('data-test', 'test');
+
+      var value = DOMPropertyOperations.getValueForProperty(
```

Comment:
We are trying to get away from unit tests like this because they make it hard to swap internals. Can you instead write an integration test? Either in the `ReactDOMServerIntegrationTest` if it's testing markup mismatch or maybe more targeted test right here (but that would use `ReactDOM.render` and reproduce this though public API instead).